### PR TITLE
Add 2018 Game Details

### PR DESCRIPTION
**Pull request by yellowjaguar5**
As noted in #10, 2018 Game Details needs to be ported from Screensteps ([article link](https://wpilib.screenstepslive.com/s/currentCS/m/getting_started/l/826278-2018-game-data-details)). Should it be located in `wpilib-overview`?

Edit: **I'll be away from the internet for two weeks (back on August 11th) and I plan to continue working on this PR when I return.**

### CODE REVIEW

**Comment by yellowjaguar5** [comment]
@Daltz333 Can you review this?

File: source/docs/software/wpilib-overview/2018-game-details.rst [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+The Game Data in LabVIEW is accessed from the Game Specific Data VI. This VI can be found in the ``WPI Robotics Library`` -> ``Driver Station`` palette.
+
+The most obvious location to query the Game Data in LabVIEW is at the beginning of the Autonomous Independent VI. You can use this information either in addition to, or in place of the "Auto Selector" code that is already present in the default template.
+
+In the example below, the String Subset VI (from the String palette) is used to grab just the first character (the direction of the Switch). This information is then used with a Case Structure (make sure to rename the cases appropriately) inside the existing Auto Selector case structure. This construct allows for Left and Right versions of multiple auto routines (e.g. Switch, Scale, Drive, etc.). The example shows the "L" case of the inner structure.
+
+.. image:: images/2018-game-details/labview.png
+
+C++
```
Comment (Daltz333) [inline source/docs/software/wpilib-overview/2018-game-details.rst]: Combine C++ and Java sections. This is what tabs are for.

File: source/docs/software/wpilib-overview/2018-game-details.rst [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    //Put left auto code here
+    } else {
+    //Put right auto code here
+    }
+  }
+
+Java
+----
+
+The Game Data in Java is accessed using the getGameSpecificMessage method in the DriverStation class. If you are using the Iterative or Timed Robot classes (or Command Based, which is based on one of those) you can query the Game Data in the AutonomousInit() method. If you are using SimpleRobot, you can query it at the beginning of your Autonomous() method.
```
Comment (Daltz333) [inline source/docs/software/wpilib-overview/2018-game-details.rst]: Methods names should be wrapped in an inline code-block

File: source/docs/software/wpilib-overview/2018-game-details.rst [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+The example below grabs the first character (the direction of the Switch) and uses an if statement with a comparison to the 'L' character to determine what code to run. Similar concepts can be extrapolated to using a switch statement instead, or combing this comparison with other means of selecting auto modes such as a Sendable Chooser.
+
+.. code-block:: java
+
+  String gameData;
+  gameData = DriverStation.getInstance().getGameSpecificMessage();
+  if(gameData.length() > 0)
```
Comment (Daltz333) [inline source/docs/software/wpilib-overview/2018-game-details.rst]: Syntax should follow the style guide, AKA: braces need to be on the same line

File: source/docs/software/wpilib-overview/2018-game-details.rst [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+Blue Alliance: "LRL"
+
+.. image:: images/2018-game-details/example-2.png
+
+Red Alliance: "RRR"
+
+Blue Alliance: "RRR"
+
+.. tabs::
```
Comment (Daltz333) [inline source/docs/software/wpilib-overview/2018-game-details.rst]: Java should be the first tab, then C++, then LabVIEW

File: source/docs/software/wpilib-overview/2018-game-details.rst [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+    The Game Data in Java is accessed using the ``getGameSpecificMessage`` method in the ``DriverStation`` class. If you are using the ``IterativeRobot`` or ``TimedRobot`` classes (or Command Based, which is based on one of those) you can query the Game Data in the ``AutonomousInit()`` method. If you are using ``SimpleRobot``, you can query it at the beginning of your ``Autonomous()`` method.
+
+    The example below grabs the first character (the direction of the Switch) and uses an if statement with a comparison to the 'L' character to determine what code to run. Similar concepts can be extrapolated to using a switch statement instead, or combing this comparison with other means of selecting auto modes such as a ``SendableChooser``.
+
+    .. code-block:: java
+
+      String gameData;
+      gameData = DriverStation.getInstance().getGameSpecificMessage();
+      if(gameData.length() > 0)
```
Comment (Daltz333) [inline source/docs/software/wpilib-overview/2018-game-details.rst]: Use same line curly brackets

File: source/docs/software/wpilib-overview/2018-game-details.rst [hunk 6]
```diff
@@ -0,0 +1,94 @@
+2018 Game Details
+=================
+
+.. note:: In the 2018 game, FIRST:sup:`®` POWER UP:sup:`SM`, the assignment of plates to alliances is randomized at the start of the match. To aid teams in programing autonomous routines, the Field Management System (FMS) will provide information to each team on the location of the plates assigned to their alliance. This page details the structure of that data and describes how to use it in each of the software languages.
```
Comment (Daltz333) [inline source/docs/software/wpilib-overview/2018-game-details.rst]:
```suggestion
.. note:: In the 2018 game, FIRST® POWER UP℠, the assignment of plates to alliances is randomized at the start of the match. To aid teams in programing autonomous routines, the Field Management System (FMS) will provide information to each team on the location of the plates assigned to their alliance. This page details the structure of that data and describes how to use it in each of the software languages.
```

**Comment by Daltz333** [comment]
We've discussed internally and don't feel like we really need this for this year. Game data wasn't applicable this past year, and will likely need to be reworked anyways if they do implement it again.

Okay with closing this PR?

**Comment by yellowjaguar5** [comment]
Sure. Thanks for letting me know. close #149.

**Pull request closed without merging**